onnx/defs/math/defs.cc

```diff
@@ -730,22 +715,28 @@ ONNX_OPERATOR_SET_SCHEMA(
           }
         }));
 
-static const char* MatMul_ver6_doc = R"DOC(
+static const char* MatMul_ver9_doc = R"DOC(
 Matrix product that behaves like numpy.matmul: https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.matmul.html
 )DOC";
 
 ONNX_OPERATOR_SET_SCHEMA(
     MatMul,
-    1,
+    9,
     OpSchema()
         .Input(0, "A", "N-dimensional matrix A", "T")
         .Input(1, "B", "N-dimensional matrix B", "T")
         .Output(0, "Y", "Matrix multiply results from A * B", "T")
         .TypeConstraint(
             "T",
-            {"tensor(float16)", "tensor(float)", "tensor(double)"},
-            "Constrain input and output types to float tensors.")
-        .SetDoc(MatMul_ver6_doc)
+            {"tensor(float16)",
```

Comment:
Shall we use all_numeric_types instead?

Reply:
no, all_numeric_types contains low precision ones there. To support low-precision, we'd have different definitions. for example, for matmul, the input may be 8 bits integer but the output should be 16-bit or 32-bit (common use) integer to handle overflow. Make sense please?